drivers/include/sdp3x.h

```diff
@@ -0,0 +1,195 @@
+    sdp3x_params_t params;          /**< Device initialization parameters */
+    mutex_t mutex;                  /**< Mutex for this sdp3x dev */
+} sdp3x_t;
+
+/**
+ * @brief   Measurement from SDP3x sensor
+ */
+typedef struct {
+    int32_t differential_pressure;  /**< Differential Pressure measured 
+                                         in centiPa */
+    int32_t temperature;            /**< Temperature measured in 100*°C */
+    int16_t dp_scale_factor;       /**< Scale factor for measurement */
+} sdp3x_measurement_t;
+
+/**
+ * @brief   Status and error return codes
+ */
+enum {
+    SDP3x_OK = 0,             /**< all went as expected */
+    SDP3x_CRCERROR = -1,      /**< CRC-Check failed */
+    SDP3x_NODATA = -2,        /**< No Data available */
+    SDP3x_IOERROR = -3,       /**< I/O error */
+    SDP3x_WRONGSIZE = -4      /**< Wrong size of array for method */
+};
+
+/** Initialize SDP3x
+ *
+ *      @param dev         sdp3x device
+ *      @param params      sdp3x device params
+ *
+ *      @return            1 if device started
+ */
+int sdp3x_init(sdp3x_t *dev, const sdp3x_params_t *params);
+
+/** Start Continuous Measuring
+ *
+ *      @param dev          sdp3x device
+ *      @param flags        stores Temperature compensation and averaging flags
+ *                          0th bit-> 0 = No averaging, 1 = Average values
+ *                          1st bit-> 0 = Differential pressure, 1 = Mass Flow
+ *
+ *      @return             1 if device started
+ */
+int8_t sdp3x_start_continuous(sdp3x_t *dev, uint8_t flags);
+
+/** read temperature and differential pressure
+ *
+ *      @param result       struct to store result
+ *      @param dev          sdp3x device
+ *
+ *      @return             1 on success
+ */
+
+int8_t sdp3x_read_continuous(sdp3x_measurement_t *result, sdp3x_t *dev);
+
+/** Read current temperature value
+ *
+ *      @param dev           sdp3x device
+ *      @param tempComp      Temperature compensation:
+ *                           1 = Mass flow, 2 = Differential pressure
+ *
+ *      @return              Current temperature in celsius multiplied by 100
+ *                           to preserve accuracy and return in integer
+ */
+int32_t sdp3x_read_single_temperature(sdp3x_t *dev, uint8_t tempComp);
+
+/** Read current differential_pressure value
+ *
+ *      @param dev           sdp3x device
+ *      @param tempComp      Temperature compensation:
+ *                           1 = Mass flow, 2 = Differential pressure
+ *
+ *      @return              current differential pressure in centipascal
+ */
+int32_t sdp3x_read_single_differential_pressure(sdp3x_t *dev,
+                                                uint8_t tempComp);
+
+/** read temperature and differential pressure
+ *
+ *      @param dev           sdp3x device
+ *      @param tempComp      Temperature compensation:
+ *                           1 = Mass flow, 2 = Differential pressure
```

Comment:
You now have a define for that, so better use it.

Reply:
Resolved